packages/node-resolve/src/index.js

```diff
@@ -267,9 +266,13 @@ export function nodeResolve(opts = {}) {
           importer,
           Object.assign({ skipSelf: true }, resolveOptions)
         );
-        const isExternal = !!(resolvedResolved && resolvedResolved.external);
-        if (isExternal) {
-          return false;
+        if (resolvedResolved) {
+          // Handle plugins that manually make the result external
+          if (resolvedResolved.external) {
+            return false;
+          }
+          // Pass on meta information added by other plugins
+          return { ...resolvedResolved, moduleSideEffects: resolved.moduleSideEffects };
```

Comment:
given what you said before

>The only things that are NOT respected are the id and moduleSideEffects

shouldn't you be forcing `resolved.id` as the id here too?

Reply:
I mean 
```suggestion
          return { ...resolvedResolved, id: resolved.id, moduleSideEffects: resolved.moduleSideEffects };
```
?

right now a change to bar.js would make it though and it shouldn’t?

Reply:
Ah, then you are indeed right, should read my own code ^^
I will change it to just pick the meta property, that is probably best for now.

Reply:
Done, please have another look